Add templatel recipe

clarete opened the pull request:
### Brief summary of what the package does

The package `templatel` provides a template language inspired by Python's Jinja library.
Although it's on its first release, it already provides most functionalities provided by Jinja.
Here's a teeny snippet of how it works:

```html
<h1>{{ title }}</h1>
<ul>
{% for user in users %}
  <li><a href="{{ user.url }}">{{ user.name }}</a></li>
{% endfor %}
</ul>
```
That can be rendered with the following Emacs Lisp code:
```emacs-lisp
(templatel-render-file "tmpl.html.jinja"
                       '(("title" . "A nice web page")
                         ("users" . ((("url" . "http://clarete.li")
                                      ("name" . "link"))
                                     (("url" . "http://gnu.org")
                                      ("name" . "Gnu!!"))))))```
```

### Direct link to the package repository

https://github.com/clarete/templatel

### Your association with the package

I'm the package's maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Quick first pass, but I see no blockers to merging this.

- templatel.el#L143: There is no need to quote lambdas (neither #' nor ')
- templatel.el#L1425: It's safer to sharp-quote function names; use `#'`
- templatel.el#L447: It's safer to sharp-quote function names; use `#'`
- templatel.el#L1153: Consider `unless ...` instead of `when (not ...)`
- templatel.el#L4: Prefer `https` over `http` if possible ([why?](https://news.ycombinator.com/item?id=22933774))
- templatel.el#L21: Prefer `https` over `http` if possible ([why?](https://news.ycombinator.com/item?id=22933774))
- Lowewr -> Lower
- paramter -> parameter

clarete commented:
Thank you so much for reviewing and merging this PR, @riscy!
I've addressed the issues you pointed out! TIL about not needing to quote lambdas!!! 😄 🎉
